### PR TITLE
P-extension PR RISCV-ISAC

### PR DESCRIPTION
**Pull request by xueliangh**
Hello Neel,

I’m the Andes engineer working on the “P” extension support for RISCV-CTG and RISCV-ISAC. We are ready to submit our initial pull request for the modification.
AS P-extension divides a register into 8/16-bit subfields for packed SIMD operations, the dataset and coverpoints become more complicated. Our first pull request will contain modifications for three instructions from the P extension only: ADD8, ADD16, and SCLIP8. We would like to seek your review first before proceed into the rest of the instructions. Please kindly review our pull request and give us guidance on how to proceed next.

Reference for P-extension, [ADD8](https://github.com/riscv/riscv-p-spec/blob/master/P-ext-proposal.adoc#add8-simd-8-bit-addition), [ADD16](https://github.com/riscv/riscv-p-spec/blob/master/P-ext-proposal.adoc#add16-simd-16-bit-addition) on pages 83-84, and [SCLIP8](https://github.com/riscv/riscv-p-spec/blob/master/P-ext-proposal.adoc#sclip8-simd-8-bit-signed-clip-value) on page 263.
https://github.com/riscv/riscv-p-spec

Hsueh-Liang

### CODE REVIEW

File: riscv_isac/InstructionObject.py [hunk 1]
```diff
@@ -72,4 +74,371 @@ def __str__(self):
             line+= ' csr_commit: '+ str(self.csr_commit)
         if self.mnemonic:
             line+= ' mnemonic: '+ str(self.mnemonic)
-        return line
\ No newline at end of file
+        return line
+
+
+    # Put the following function in instructionObject.py 
+    def init_rvp_dictionary(self):
```
Comment (pawks) [inline riscv_isac/InstructionObject.py:81]: Please move these functions to the `internal_decoder.py`. The `InstructionObject` is supposed to describe the instruction with the values for all the fields in the instruction. All decoding logic should be isolated to the plugin and not creep up in the `InstructionObject`.

File: riscv_isac/cgf_normalize.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ def twos(val,bits):
     :result: two's complement version of the input
 
     '''
-    if isinstance(val,str):
+    if isinstance(val,str)
```
Comment (pawks) [inline riscv_isac/cgf_normalize.py:21]: Incorrect Syntax.

File: riscv_isac/plugins/internaldecoder.py [hunk 3]
```diff
@@ -341,6 +342,197 @@ def arithi_ops(self, instrObj):
 
         return instrObj
 
+    # Automatically generated by parse_opcodes.
+
+    # Put the following function in internaldecoder.py 
+    def rvp_ops(self, instrObj):
```
Comment (pawks) [inline riscv_isac/plugins/internaldecoder.py:348]: The `rvp` specific dictionaries can probably be defined as a part of the decoder class over here.

**Review by pawks — CHANGES_REQUESTED** [review]

**Comment by neelgala** [comment]
Also an entry in the Changelog.md and a bumpversion is required.

File: riscv_isac/cgf_normalize.py (hunk 2 above)
Comment (yungchinghsiao) [reply riscv_isac/cgf_normalize.py]: This was an accidental overwrite. This line has been restored to the version in 17ff031e by rewriting the history of this branch.

File: riscv_isac/plugins/internaldecoder.py (hunk 3 above)
Comment (yungchinghsiao) [reply riscv_isac/plugins/internaldecoder.py:713]: Done in 571c5525724b34f23a346e97628a1b1bc144be59

File: riscv_isac/InstructionObject.py (hunk 1 above)
Comment (yungchinghsiao) [reply riscv_isac/InstructionObject.py]: Done in 571c5525724b34f23a346e97628a1b1bc144be59

**Review by pawks — APPROVED** [review]

**Pull request merged**